MdeModulePkg/Core/Dxe/Misc/MemoryProtectionSupport.c

```diff
@@ -1655,9 +1655,43 @@ SyncBitmapAndSetAccessAttributesInMemoryMap (
     // If attributes are nonzero, set the corresponding bit in the bitmap
     if (MemoryMapEntry->Attribute != 0) {
       SET_BITMAP_INDEX (Bitmap, Index);
+    }
+
+    Index++;
+    MemoryMapEntry = NEXT_MEMORY_DESCRIPTOR (MemoryMapEntry, *DescriptorSize);
+  }
+}
 
-      // Skip code regions
-    } else if (!IsCodeType (MemoryMapEntry->Type)) {
+/**
+  Set access attributes in the memory map based on the memory protection policy and
+  mark visited regions in the bitmap.
+
+  @param[in]        MemoryMapSize       A pointer to the size, in bytes, of the
+                                        MemoryMap buffer.
+  @param[in, out]   MemoryMap           A pointer to the current memory map.
+  @param[in]        DescriptorSize      Size, in bytes, of an individual EFI_MEMORY_DESCRIPTOR.
+  @param[in]        Bitmap              Pointer to the beginning of the bitmap to be updated
+**/
+STATIC
+VOID
+SetAccessAttributesInMemoryMap (
+  IN CONST UINTN             *MemoryMapSize,
+  IN  EFI_MEMORY_DESCRIPTOR  *MemoryMap,
+  IN CONST UINTN             *DescriptorSize,
+  IN CONST UINT8             *Bitmap
+  )
+{
+  EFI_MEMORY_DESCRIPTOR  *MemoryMapEntry;
+  EFI_MEMORY_DESCRIPTOR  *MemoryMapEnd;
+  UINTN                  Index = 0;
+
+  MemoryMapEntry = MemoryMap;
+  MemoryMapEnd   = (EFI_MEMORY_DESCRIPTOR *)((UINT8 *)MemoryMap + *MemoryMapSize);

```

Comment:
Do we need to check MemoryMapSize isn't null before dereferencing or we are confident that the callers populated it correctly?

Reply:
I see this is only called in one place below, where it is checked as input, so seems we are confident (CodeQL still might complain).

Reply:
Added NULL checks for all input paramters